pkg/applicationmanager/controllers/applicationregistration.go

```diff
@@ -170,13 +171,25 @@ func RegisterDefaultCRDs() error {
 			if err := registerCRDV1(*crd); err != nil {
 				logrus.WithError(err).Error("unable to create appreg for v1 crd")
 			}
+			if len(crd.Spec.Names.Kind) > 0 {
+				if _, ok := genericcache.Instance().GetCRDkindToPluralmap()[strings.ToLower(crd.Spec.Names.Kind)]; !ok {
+					genericcache.Instance().SetCRDkindToPluralmap(crd.Spec.Names.Kind, crd.Spec.Names.Plural)
```

Comment:
nit..CRDkindToPluralmap-> Kind..Map. Let's try to follow camel case for all functions and variables.
SetCRDkindToPluralmap, crdkindToPluralmap